controllers/datadogagent/feature/apm/rbac.go

```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package apm
+
+import (
+	"github.com/DataDog/datadog-operator/pkg/kubernetes/rbac"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+func getLanguageDetectionRBACPolicyRules() []rbacv1.PolicyRule {
+	rbacRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{rbac.AppsAPIGroup},
+			Resources: []string{
+				// Currently only deployments are supported
```

Comment:
Note: if we want to avoid modifying the operator everytime we add a supported resource, it's better to add them all

Reply:
I think it would be better to stick to the minimum privilege principle, we should not add a permission that is not currently used. 

To add, I assume adding support for new workload types will be done in the same release, so it would be needed to update the operator only once. 

And finally, it is better to keep the operator coherent with helm. 

WDYT?

Reply:
No strong opinion here. IMO we'll add the support very soon and it can be confusing for some users that might upgrade the agent and not the operator